emcc.py

```diff
@@ -1369,6 +1369,18 @@ def default_setting(name, new_default):
   if settings.DISABLE_EXCEPTION_THROWING and not settings.DISABLE_EXCEPTION_CATCHING:
     exit_with_error("DISABLE_EXCEPTION_THROWING was set (probably from -fno-exceptions) but is not compatible with enabling exception catching (DISABLE_EXCEPTION_CATCHING=0). If you don't want exceptions, set DISABLE_EXCEPTION_CATCHING to 1; if you do want exceptions, don't link with -fno-exceptions")
 
+  # SUPPORT_LONGJMP=1 means SUPPORT_LONGJMP=emscripten (backward compatibility)
+  if settings.SUPPORT_LONGJMP == 0:
+    settings.SUPPORT_LONGJMP = 'none'
+  if settings.SUPPORT_LONGJMP == 1:
+    settings.SUPPORT_LONGJMP = 'emscripten'
+  # Wasm SjLj cannot be used with Emscripten EH
+  if not settings.DISABLE_EXCEPTION_CATCHING and settings.SUPPORT_LONGJMP == 'wasm':
+    exit_with_error('SUPPORT_LONGJMP=wasm cannot be used with DISABLE_EXCEPTION_CATCHING=0')
+  # DISABLE_EXCEPTION_THROWING=0 is the default, so we change it here
+  if settings.SUPPORT_LONGJMP == 'wasm':
+    settings.DISABLE_EXCEPTION_THROWING = 1
```

Comment:
This last if surprises me. So for now we don't support wasm EH + wasm Sjlj?

If so should this error instead?

Reply:
I agree it's ok as its experimental. Still I think it's more consistent with other stuff to error on combinations we don't support. That might save confusion later. So I'd suggest
```suggestion
    if settings.DISABLE_EXCEPTION_THROWING == 0:
      exit_with_error('wasm sjlj is not supported with emscripten exceptions (and will not be supported)')
    if settings.EXCEPTION_HANDLING == 0:
      exit_with_error('wasm sjlj is not supported with wasm exceptions yet')
```